src/mine/api_client.cpp

```diff
@@ -40,18 +41,53 @@ ApiClient& ApiClient::getDefaultInstance() {
   return api_client;
 }
 
+std::string ApiClient::serialize(std::string str) {
+  std::string result;
+  for (char c : str) {
+    if (c == '\\') {
+      result += "\\\\";
+    } else if (c == '\"') {
+      result += "\\\"";
+    } else if (c == '\n') {
+      result += "\\n";
+    } else if (c == '\r') {
+      // Ignore CR
+    } else {
+      result += c;
+    }
+  }
+  return result;
+}
+
+std::string ApiClient::toJson(const Program& program){
+  const std::string id = Comments::getSequenceIdFromProgram(program);
+  const std::string submitter = Comments::getSubmitter(program);
+  const std::string change_type = Comments::getCommentField(program, Comments::PREFIX_CHANGE_TYPE);
+  const std::string mode = (change_type == "" ? "add" : "update");
```

Comment:
The change_type is set to the `status` field in the checker result. For new programs it is "Found" (the first word in miner bot messages in Discord): https://github.com/loda-lang/loda-cpp/blob/3887bf65807d1307831407cca0cf881b1b305bf8/src/mine/checker.cpp#L206

So you should map "Found" => "new" and otherwise => "update".